src/libponyc/expr/operator.c

```diff
@@ -480,6 +480,18 @@ bool expr_assign(pass_opt_t* opt, ast_t* ast)
       }
     }
 
+    if(ast_child(left) != NULL)
+    {
+      ast_error(opt->check.errors, ast,
+        "not safe to write right side to left side");
+      ast_error_continue(opt->check.errors, r_type, "right side type: %s",
+        ast_print_type(r_type));
+      ast_error_continue(opt->check.errors, ast_child(left), "left side type: %s",
+        ast_print_type(ast_type(ast_child(left))));
+      ast_free_unattached(wl_type);
+      return false;
+    }
```

Comment:
this shouldn't be its own block that duplicates most of the code below. The "add a check for left side and output accordingly" can be done at line 487 of the existing code.

Reply:
Done